docs/readmes/proposals/p014_nms_regression_testing.md

```diff
@@ -0,0 +1,116 @@
+---
+id: p014_nms_regression_test
+title: NMS Regression Testing
+hide_title: true
+---
+
+# Proposal: NMS Regression Testing
+
+Author(s): @andreilee
+
+Last updated: 04/20/2021
+
+Discussion at
+[https://github.com/magma/magma/issues/4888](https://github.com/magma/magma/issues/4888).
+
+## Abstract
+
+
+The Magma NMS currently lacks many end-to-end tests for critical worksflows.
+Given the current velocity of contributions to the Magma NMS, a tool such as
+[Testim](https://www.testim.io/) will be used and integrated with Magma's
+CI framework to cover the gap in testing, with limited time investment from
+involved developers. 
+
+## Background
+
+As of the time of this document's conception, Magma has high feature velocity.
+Without infinite developer resources, this presents a trade-off between
+spending time on feature work and validation.
+The use of tools such as Testim are intended to free up time for developers to
+focus on feature work.
+
+## Proposal
+
+Testim will be used for daily testing of various workflows on NMS.
+This testing will be limited to what can be automatically tested without
+interaction with eNodeBs, CPEs, or UE devices, and will focus on interactivity
+with the Magma NMS.
+
+A representative test set of NMS workflows useful for verification of pure-play
+FWA usage of Magma will be created on Testim.
+The Testim account will be managed by the Magma Core Foundation, and the
+subscription will be paid for by Facebook.
+
+The test suite will be triggered to run daily through Github Actions.
+A kubernetes cluster will be set up to run daily images of the Magma
+orchestrator and NMS. The tests will run on this test cluster.
+
+**Daily Reports**
+
+A Slack channel will be used for contributors to see the results of tests.
+The results of each test should be visible in the Slack channel, and a link
+to the test run should also be provided, as well as a reference to which
+container image tags were used for the test run.
+
+**Budget**
+
+Facebook will pay for the associated costs with using the service provided by
+Testim, for at least one year.
+
+**Tests**
+
+The following tests are proposed to verify pure-play FWA usage of the Magma
+NMS:
+
+1. Create/update/delete an LTE network
+2. Create/update/delete an LTE AGW
```

Comment:
Variants to be tested here - gateways with different IP allocation schemes including IP_POOL, bridged configuration and STATIC_IP configuration